### ISSUE TITLE
Misconfigured mTLS supportability issue and leads to Internal Server Error

### ISSUE THREAD

**Issue by flands** [open-telemetry/opentelemetry-collector#2061]
**Describe the bug**
When attempting to export over gRPC OTLP via mTLS to a destination not properly configured, client eventually receives Internal Server Error.

**Steps to reproduce**

Agent

```
extensions:
  zpages:
    endpoint: localhost:55681

receivers:
  zipkin:

processors:

exporters:
  logging:
  otlp:
    endpoint: myserver.local:55690
    insecure: false
    ca_file: server.crt
    cert_file: client.crt
    key_file: client.key

service:
  extensions: [zpages]
  pipelines:
    traces:
      receivers: [zipkin]
      processors: []
      exporters: [logging, otlp]
```

Gateway (notice `tls_settings` is not configured for the receiver)
```
extensions:
  zpages:
    endpoint: localhost:55691

receivers:
  otlp:
    protocols:
      grpc:
        endpoint: mysite.local:55690

processors:

exporters:
  logging:

service:
  extensions: [zpages]
  pipelines:
    traces:
      receivers: [otlp]
      processors: []
      exporters: [logging]
```

Curl the Agent with: `while (true); do curl -X POST localhost:9411/api/v2/spans -H'Content-Type: application/json' -d @trace.json; done`

Wait...

**What did you expect to see?**
A clear error message that something was wrong -- either a log of something in zpages.

**What did you see instead?**
No clear error message... eventually client is returned `Internal Server Error` and zpages error sample shows:

```
2020/11/04-15:33:53.732952      .   239 trace_id: 42e8f31f67ff37af13a2980933f22c43 span_id: fbaa5e762b6a0561
                                        Status{canonicalCode=UNKNOWN, description="failed to add to the queue"}
                                        Attributes:{accepted_spans=0, format="zipkinV2", refused_spans=1, transport="http_v2_json"}
```

In short two major issues:
1) No indication to the user what the problem is or how to fix it
2) 500 error returned to the client

**What version did you use?**
0.13.0 / master

**What config did you use?**
See above

**Environment**
Mac OSX

**Additional context**

**Comment by flands** [comment]
FYI -- further testing indicates this problem is not isolated mtls. Here is how to easily reproduce:

```
exporters:
  zipkin:
    endpoint: "https://httpstat.us/400"
    retry_on_failure:
      enabled: false
    sending_queue:
      enabled: false
```
